src/Cardano/CoinSelection/LargestFirst.hs

```diff
@@ -20,77 +21,196 @@ import Cardano.Types
 import Control.Arrow
     ( left )
 import Control.Monad
-    ( foldM, when )
+    ( foldM )
 import Control.Monad.Trans.Except
     ( ExceptT (..), except, throwE )
 import Data.Functor
     ( ($>) )
 import Data.List.NonEmpty
     ( NonEmpty (..) )
 import Data.Ord
-    ( Down (..), comparing )
+    ( Down (..) )
 
 import qualified Data.List as L
 import qualified Data.List.NonEmpty as NE
 import qualified Data.Map.Strict as Map
 
--- | Implements the __Largest-First__ coin selection algorithm.
+-- | Generate a coin selection according to the __largest first__ algorithm.
+--
+-- === Summary
+--
+-- For the given /output list/ and /initial UTxO set/, this algorithm generates
+-- a /coin selection/ that is capable of paying for all of the outputs, and a
+-- /remaining UTxO set/ from which spent values have been removed.
```

Comment:
```suggestion
-- /remaining UTxO set/ from which all spent values have been removed.
```

Reply:
Fixed in https://github.com/input-output-hk/cardano-coin-selection/pull/7/commits/aae8f10ba8cd883dbed18881376d293b47c6133e.